applications/nfc/scenes/nfc_scene_mifare_desfire_app.c

```diff
@@ -0,0 +1,109 @@
+#include "../nfc_i.h"
+
+#define TAG "NfcSceneMifareDesfireApp"
+
+enum SubmenuIndex {
+    SubmenuIndexAppInfo,
+    SubmenuIndexDynamic, // dynamic indexes start here
+};
+
+MifareDesfireApplication* nfc_scene_mifare_desfire_app_get_app(Nfc* nfc) {
+    uint32_t app_idx =
+        scene_manager_get_scene_state(nfc->scene_manager, NfcSceneMifareDesfireApp) >> 1;
+    MifareDesfireApplication* app = nfc->dev->dev_data.mf_df_data.app_head;
+    for(int i = 0; i < app_idx && app; i++) {
+        app = app->next;
+    }
+    return app;
+}
+
+void nfc_scene_mifare_desfire_app_submenu_callback(void* context, uint32_t index) {
+    Nfc* nfc = (Nfc*)context;
+
+    view_dispatcher_send_custom_event(nfc->view_dispatcher, index);
+}
+
+void nfc_scene_mifare_desfire_app_on_enter(void* context) {
+    Nfc* nfc = (Nfc*)context;
+    Submenu* submenu = nfc->submenu;
+    MifareDesfireApplication* app = nfc_scene_mifare_desfire_app_get_app(nfc);
+    if(!app) {
+        FURI_LOG_E(TAG, "Bad state. No app selected?");
+        return;
```

Comment:
If this codepath ever runs, it's a bug in the MifareDesfireData scene.  But just in case I made a nice popup anyway.  :)  6beecbaab9ceb871117c0a772cf7db890c0b24dd

![error_popup](https://user-images.githubusercontent.com/184975/158894894-d8396307-eafb-4a37-be86-43ec3d7480bd.png)